safe/gui/widgets/dock.py

```diff
@@ -560,6 +565,43 @@ def validate(self):
             message = self.ready_message()
             return True, message
 
+    @pyqtSlot(QgsMapLayer, str)
+    def save_auxiliary_files(self, layer, destination):
+        """Save auxiliary files when using the saved as function.
+
+        If some auxiliary files (.xml or .keywords) exist, this function will copy them
+        when the saved as function is used on the layer.
+
+        :param layer: The layer which has been saved as.
+        :type layer: QgsMapLayer
+
+        :param destination: The new filename of the layer
+        :type str
+
+        """
+
+        source_basename = os.path.splitext(layer.source())[0]
+        source_keywords = "%s.keywords" % source_basename
+        source_xml = "%s.xml" % source_basename
+
+        destination_basename = os.path.splitext(destination)[0]
+        destination_keywords = "%s.keywords" % destination_basename
+        destination_xml = "%s.xml" % destination_basename
+
+        #Keywords
+        if os.path.isfile(destination_keywords):
```

Comment:
probably you need to add an exception handler for file permissions errors here and similar clauses below.